### ISSUE TITLE
switching filter rule - 'list' object has no attribute 'keys'

### ISSUE THREAD

**Issue by toddstar** [travisghansen/hass-opnsense#10]
I have a couple cheap ebay/amazon camera that i've blocked from accessing the internet via a LAN rule in opnsense (using alias for ip matching them) to stop them calling home / disable the external remote access apps, trying to setup an automation to briefly toggle the rule (so that they can NTP sync) but it's failing with `'list' object has no attribute 'keys'` on sending switch turn off (and switch turn on). Can see in opnsense that the rule is being disabled by the calls, and its working as can see timestamp being corrected on the cameras, however as it errors its stopping the automation on turn off and not doing the delay & turn back on part

![Screenshot 2021-11-21 222436](https://user-images.githubusercontent.com/22575856/142781279-a8d02a20-bc4d-4689-8ec5-0c7fa08639c2.png)

![Screenshot 2021-11-21 222816](https://user-images.githubusercontent.com/22575856/142781344-d006120e-d758-4964-9cd0-f6a1d2e6b835.png)

![Screenshot 2021-11-21 222827](https://user-images.githubusercontent.com/22575856/142781350-037f135b-413c-411f-9d81-d49cffeedda7.png)

**log errors**

```
This error originated from a custom integration.

Logger: homeassistant
Source: custom_components/opnsense/sensor.py:231
Integration: OPNsense (documentation, issues)
First occurred: 22:36:30 (1 occurrences)
Last logged: 22:36:30

Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 134, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 265, in _async_refresh
    update_callback()
  File "/config/custom_components/opnsense/__init__.py", line 354, in process_entities
    entities = self.process_entities_callback(self.hass, self.config_entry)
  File "/config/custom_components/opnsense/sensor.py", line 231, in process_entities_callback
    for gateway_name in state["telemetry"]["gateways"].keys():
AttributeError: 'list' object has no attribute 'keys'

```

```
Logger: homeassistant.components.automation.toggle_camera_wan
Source: custom_components/opnsense/sensor.py:231
Integration: Automation (documentation, issues)
First occurred: 22:32:22 (2 occurrences)
Last logged: 22:32:22

toggle camera WAN: Error executing script. Unexpected error for call_service at pos 1: 'list' object has no attribute 'keys'
While executing automation automation.toggle_camera_wan
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 371, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 571, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1495, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1530, in _execute_service
    await handler.job.target(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    await self.hass.helpers.service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 667, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 863, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 704, in _handle_entity_call
    await result
  File "/config/custom_components/opnsense/switch.py", line 268, in async_turn_off
    await self.coordinator.async_refresh()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 165, in async_refresh
    await self._async_refresh(log_failures=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 265, in _async_refresh
    update_callback()
  File "/config/custom_components/opnsense/__init__.py", line 354, in process_entities
    entities = self.process_entities_callback(self.hass, self.config_entry)
  File "/config/custom_components/opnsense/sensor.py", line 231, in process_entities_callback
    for gateway_name in state["telemetry"]["gateways"].keys():
AttributeError: 'list' object has no attribute 'keys'
```

**Comment by travisghansen** [comment]
Welcome! How are people even finding this integration? lol

I literally just a pushed a fix of similar nature but I don't think it's what you're hitting (https://github.com/travisghansen/hass-opnsense/commit/e4a892e191a81ab50d6bd2496289106e7de915fc) (it's totally unrelated to filter rules).

Yup, with your edit you confirmed, the most recent commit should fix that.

**Comment by toddstar** [comment]
I saw your pfsense post on HA community forum and thought wonder if that'd work for opnsense, came back a few days later (to ask) and saw the conversation with alexdelprete and followed the trail lol.

Thanks!

**Comment by travisghansen** [comment]
Did the update fix your issue?

**Comment by toddstar** [comment]
yeah working fine now! 

cheers

![Screenshot 2021-11-21 2254](https://user-images.githubusercontent.com/22575856/142782175-5822ce26-e938-42ad-9c84-bd1a5d6a9f9b.png)

**Comment by alexdelprete** [comment]
> Welcome! How are people even finding this integration? lol

I told you OPNsense users were waiting since a long time for this...:)

**Comment by travisghansen** [comment]
Glad it’s working! I’m amazed at the automations that people come up with. Cool stuff.

**Comment by alexdelprete** [comment]
> yeah working fine now!
> 
> cheers
> 
> ![Screenshot 2021-11-21 2254](https://user-images.githubusercontent.com/22575856/142782241-1f92f4e8-f133-4539-8525-a48c0847b4f8.png)

Todd, is this the NR debugger or the HA one?

**Comment by travisghansen** [comment]
That’s the automation debugger in HA.

**Comment by alexdelprete** [comment]
> That’s the automation debugger in HA.

I need to start using it...still debugging things without it. :)

**Comment by alexdelprete** [comment]
> trying to setup an automation to briefly toggle the rule (so that they can NTP sync)

Todd, regarding this NTP thing: in my OPNsense I forcefully redirect every clients' DNS/NTP requests to WAN to local DNS/NTP servers, through Port Forward rules:

![image](https://user-images.githubusercontent.com/7027842/142784583-8fbfd131-534b-4267-bcca-86b3a66afee6.png)

**Comment by toddstar** [comment]
never thought of trying that so will give it a go, although somewhat thinking a firmware update and it not being able to connect home is causing my issues with one of the cameras to keep resetting rather than time being out of sync; rtsp stream was solid for a good 16 months with net access blocked via my old ISP router & time being completely wrong. Whilst removing the ISP kit it was allowed to access the web for 15 mins and since then it been throwing itself off the wifi at an exponential daily rate.

**Comment by alexdelprete** [comment]
As a general approach: I don't allow any client to use external services like DNS/NTP when those are offered locally.

Regarding the many devices coming from many manufacturers that offer poor documentation (mainly chinese stuff, even branded devices), I found out they do strange things that are not documented nor understandable. So I don't allow those devices access to WAN in general, if they need DNS/NTP, I redirect them to local services like I showed you, they don't even know the answer to their requests come from local servers, it works fine.
